feat: Add password reset

rudzzz opened the pull request:
# DSD-87 Add password reset

### **Changes**  

#### **Password Reset**  
- Users can reset their password using the `/reset-password` page.  
- Supabase ensures that only authenticated users (via a valid reset link) can update their password.  
- After successfully setting a new password, the user is forcefully logged out to prevent session inconsistencies.  
- They must then log in again with their new credentials.  
- Displays appropriate success and error messages throughout the process.  

#### **Middleware**  
- Previously, authentication logic only differentiated between auth pages and public pages, but it was blocking Supabase’s reset password redirection, preventing users from completing the process.
- Added a `PROTECTED_PAGES` array to define restricted areas (`/account`, `/dashboard`, `/schedule`).  
- Now, if an **unauthenticated user** attempts to access a protected page, they are redirected to `/login`.  
- If an **authenticated user** tries to access authentication-related pages (like `/login` or `/register`), they are redirected to `/account`.

rudzzz committed:
feat: Add password reset

katiehom reviewed (changes requested):
Acir - this looks good, but one thing I would want added is that after a user changes their password having a toast notification that it was successful or not and then redirect them back to the login page. I tested it out and wasn't sure if it changed or not until I went back to login and the new password works. So -- the hookup is awesome, just that redirection afterwards!

katiehom commented:
https://github.com/user-attachments/assets/d0da4ae0-c346-4c5b-a749-676c9a9e4c2c

destocot reviewed (commented):
you can use createClient from /utils/supabase/client or /utils/supabase/server
no need to import from supabase-js or ssr

the reset password should be handled through the account page

you removed the redirection for isAuthenticated and is on auth page
then a user can go to login page if they are logged in already which is not bad but it is not the best UX

rudzzz committed:
Merge branch 'dev' of https://github.com/Demonico/supreme-bassoon into feat/DSD-87-password-reset

rudzzz committed:
Merge branch 'dev' of https://github.com/Demonico/supreme-bassoon into feat/DSD-87-password-reset

rudzzz committed:
feat: Update password and toast notifications

rudzzz commented:
## Changes 

- Implemented a password update feature for authenticated users.
- Added toast notifications to provide feedback on success and errors
- Implemented redirection for unauthenticated users attempting to access protected pages.

katiehom reviewed (approved):
Acir - this looks great! I like the addition of the "Change password" button on the user account page, the addition of the toast notifications, and the "confirm password" when changing!

There was only one little hiccup that I found which doesn't need to be addressed for this PR to be approved, in my opinion, and that is that when the change password dialog is up that it might be good to prevent the dialog from closing with outside clicks - I accidentally clicked the screen to stop a notification of my password-saving software and the dialog closed. 

Nice work!

rudzzz committed:
Merge branch 'dev' of https://github.com/Demonico/supreme-bassoon into feat/DSD-87-password-reset

rudzzz committed:
Merge branch 'dev' of https://github.com/Demonico/supreme-bassoon into feat/DSD-87-password-reset

rudzzz committed:
Merge branch 'dev' of https://github.com/Demonico/supreme-bassoon into feat/DSD-87-password-reset

rudzzz committed:
fix: Fix reset password bug

rudzzz commented:
### Changes:

- **Implemented `route.ts` to handle password reset requests via Supabase.**  
  - This new API route processes `POST` requests, validates the input, updates the user's password via Supabase, and handles potential errors.  
  - Keeping the logic entirely on the server improves security and eliminates unnecessary client-side operations.

- **Removed `reset-password.ts` as it was redundant.**  
  - The `reset-password.ts` function previously acted as a client-side fetch wrapper for the API route, adding an unnecessary layer.  
  - Now that the API route handles everything directly, this function was no longer needed.
